README.md

```diff
@@ -14,36 +14,133 @@
+
+## Documentation
 
-See our [CHANGELOG][chaneglog_file].
+Complete documentation is available on [Read the Docs][link_rtfd]. This includes:
+- API Reference
+- Usage Examples
+- Integration Guides
+- Best Practices
+
+## Responsibilities
+
+* Provide a general-purpose Python implementation of [CycloneDX][link_website]
+* Provide type hints for implementation support
+* Support JSON/XML document parsing and generation
+* Validate CycloneDX documents against schema specifications
+* Support multiple CycloneDX specification versions
+* Maintain comprehensive data models for BOM manipulation
+* Enable pip-based installation for downstream usage
+
+## Capabilities
+
+### Enums
+* `BomFormat` - BOM format types
+* `ComponentType` - Types of components (e.g., APPLICATION, LIBRARY)
+* `ComponentScope` - Component scope types
+* `DataFlow` - Data flow types
+* `Encoding` - Encoding types
+* `ExternalReferenceType` - Types of external references
+* `HashAlgorithm` - Supported hash algorithms
+* `ImpactAnalysisAffectedStatus` - Impact analysis affected status types
+* `ImpactAnalysisJustification` - Impact analysis justification types
+* `ImpactAnalysisResponse` - Impact analysis response types
+* `ImpactAnalysisState` - Impact analysis state types
+* `IssueClassification` - Issue classification types
+* `LifecyclePhase` - Lifecycle phase types
+* `PatchClassification` - Patch classification types
+* `VulnerabilityScoreSource` - Vulnerability score source types
+* `VulnerabilitySeverity` - Vulnerability severity types
+
+### Data Models
+
+#### Core Models
+* `Bom` - Core BOM model
+* `BomRef` - BOM reference handling
+* `Metadata` - BOM metadata
+
+#### Component & Service Models
+* `Component` - Component representation
+* `ComponentEvidence` - Component evidence data
+* `Service` - Service representation
+
+#### Dependency Models
+* `Dependency` - Dependency information
+* `DependencyGraph` - Dependency relationships
+
+#### License Models
+* `License` - Base license model
+* `LicenseExpression` - License expression handling
+* `NamedLicense` - Named license representation
+* `SpdxLicense` - SPDX license support
+
+#### Analysis Models
+* `ImpactAnalysis` - Impact analysis data
+* `Issue` - Issue tracking
+* `Vulnerability` - Vulnerability information
+
+#### Reference & Organization Models
+* `ExternalReference` - External reference data
+* `Hash` - Hash information
+* `OrganizationalContact` - Contact information
+* `OrganizationalEntity` - Organization information
+
+#### Management Models
+* `Property` - Property handling
+* `Tool` - Tool representation
+
+#### Repository Models
+* `BomRefRepository` - BOM reference management
+* `ComponentRepository` - Component management
+* `ExternalReferenceRepository` - External reference management
+* `LicenseRepository` - License management
+* `PropertyRepository` - Property management
+* `ToolRepository` - Tool management
+
+### Utilities
+* Serial number generation for BOMs
+* Hash calculation helpers
```

Comment:
we do have this, but we dont want to brag about it, these are mere helpers for experts. we better not tell about them here.
please remove